src/zenml/config/global_config.py

```diff
@@ -433,9 +434,14 @@ def copy_active_configuration(
         # if a custom load config path is specified, use it to replace the
         # current store local path in the profile URL
         if load_config_path:
-            profile.store_url = store.url.replace(
-                str(config_copy.config_directory), load_config_path
+            store_path = store.get_path_from_url(store.url)
+            assert store_path
+
+            relative_store_path = PurePath(store_path).relative_to(
+                config_copy.config_directory
             )
+            new_store_path = str(load_config_path / relative_store_path)
+            profile.store_url = store.get_local_url(new_store_path)
```

Comment:
```suggestion
            store_path = store.get_path_from_url(store.url)
            if store_path:
                relative_store_path = PurePath(store_path).relative_to(
                config_copy.config_directory
            )
                new_store_path = str(load_config_path / relative_store_path)
                profile.store_url = store.get_local_url(new_store_path)
            else:
                profile.store_url = store.url
            
```

This doesn't work if you're using a profile pointing to a ZenServer. You can just use `store.url` unaltered as `profile.store_url`, if `get_path_from_url` returns None here.

Reply:
Nice catch, thank you!